src/field/social/social_buttons_pane.jsx

```diff
@@ -6,6 +6,16 @@ import { logIn } from '../../quick-auth/actions';
 import { displayName, socialConnections, authButtonsTheme } from '../../connection/social/index';
 
 export default class SocialButtonsPane extends React.Component {
+  handleAuthButtonClick(lock, provider, isSignUp) {
+    const prov = typeof provider !== 'object' ? provider.toJS() : provider;
```

Comment:
You can have multiples connection names per strategy for enterprise (for social it might not be that common) But with connection name and strategy will be enough.